xorg.xf86videovmware is broken

### Describe the bug

```
building '/nix/store/vpa89ysd4r467s0ry4rq2a1kg9qx17b0-xf86-video-vmware-13.3.0.drv'...
unpacking sources
unpacking source archive /nix/store/l1byfd2ds9r3qr0fxsz8q7ngczqvdlcl-xf86-video-vmware-13.3.0.tar.bz2
source root is xf86-video-vmware-13.3.0
setting SOURCE_DATE_EPOCH to timestamp 1526542955 of file xf86-video-vmware-13.3.0/INSTALL
patching sources
configuring
fixing libtool script ./ltmain.sh
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/hjs641iywah5swl982msq621kzzap579-xf86-video-vmware-13.3.0
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for a BSD-compatible install... /nix/store/fvprxgcxf4px865gdjd81fbwnxcjrg41-coreutils-9.0/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /nix/store/fvprxgcxf4px865gdjd81fbwnxcjrg41-coreutils-9.0/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking whether make supports nested variables... yes
checking dependency style of gcc... none
checking for gcc option to accept ISO C99... none needed
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /nix/store/lhambyc1v2c7qzzr5sq7p449xs1j6pg8-gnugrep-3.7/bin/grep
checking for egrep... /nix/store/lhambyc1v2c7qzzr5sq7p449xs1j6pg8-gnugrep-3.7/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking whether __clang__ is declared... no
checking whether __INTEL_COMPILER is declared... no
checking whether __SUNPRO_C is declared... no
checking pkg-config is at least version 0.9.0... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a sed that does not truncate output... /nix/store/9mp06ni69a44dmrjhn28mn15brdry52w-gnused-4.8/bin/sed
checking if gcc supports -Werror=unknown-warning-option... no
checking if gcc supports -Werror=unused-command-line-argument... no
checking if gcc supports -Wall... yes
checking if gcc supports -Wpointer-arith... yes
checking if gcc supports -Wmissing-declarations... yes
checking if gcc supports -Wformat=2... yes
checking if gcc supports -Wstrict-prototypes... yes
checking if gcc supports -Wmissing-prototypes... yes
checking if gcc supports -Wnested-externs... yes
checking if gcc supports -Wbad-function-cast... yes
checking if gcc supports -Wold-style-definition... yes
checking if gcc supports -Wdeclaration-after-statement... yes
checking if gcc supports -Wunused... yes
checking if gcc supports -Wuninitialized... yes
checking if gcc supports -Wshadow... yes
checking if gcc supports -Wmissing-noreturn... yes
checking if gcc supports -Wmissing-format-attribute... yes
checking if gcc supports -Wredundant-decls... yes
checking if gcc supports -Wlogical-op... yes
checking if gcc supports -Werror=implicit... yes
checking if gcc supports -Werror=nonnull... yes
checking if gcc supports -Werror=init-self... yes
checking if gcc supports -Werror=main... yes
checking if gcc supports -Werror=missing-braces... yes
checking if gcc supports -Werror=sequence-point... yes
checking if gcc supports -Werror=return-type... yes
checking if gcc supports -Werror=trigraphs... yes
checking if gcc supports -Werror=array-bounds... yes
checking if gcc supports -Werror=write-strings... yes
checking if gcc supports -Werror=address... yes
checking if gcc supports -Werror=int-to-pointer-cast... yes
checking if gcc supports -Werror=pointer-to-int-cast... yes
checking if gcc supports -pedantic... yes
checking if gcc supports -Werror... yes
checking if gcc supports -Werror=attributes... yes
Package xorg-macros was not found in the pkg-config search path.
Perhaps you should add the directory containing `xorg-macros.pc'
to the PKG_CONFIG_PATH environment variable
No package 'xorg-macros' found
checking whether make supports nested variables... (cached) yes
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /nix/store/9mp06ni69a44dmrjhn28mn15brdry52w-gnused-4.8/bin/sed
checking for fgrep... /nix/store/lhambyc1v2c7qzzr5sq7p449xs1j6pg8-gnugrep-3.7/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/fvprxgcxf4px865gdjd81fbwnxcjrg41-coreutils-9.0/bin/dd
checking how to truncate binary pipes... /nix/store/fvprxgcxf4px865gdjd81fbwnxcjrg41-coreutils-9.0/bin/dd bs=4096 count=1
./configure: line 13375: /usr/bin/file: No such file or directory
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking if RANDR is defined... yes
checking if RENDER is defined... yes
checking if DPMSExtension is defined... yes
checking if XINERAMA is defined... yes
checking if XV is defined... yes
checking for XORG... yes
checking for X11... yes
checking whether XSERVER_LIBPCIACCESS is declared... yes
checking for PCIACCESS... yes
checking for LIBDRM... yes
checking for XATRACKER... yes
checking whether to build Kernel Mode Setting and 3D... yes
checking for LIBUDEV... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating man/Makefile
config.status: creating saa/Makefile
config.status: creating vmwgfx/Makefile
config.status: creating src/Makefile
config.status: creating vmwarectrl/Makefile
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
building
build flags: -j4 -l4 SHELL=/nix/store/l0wlqpbsvh1pgvhcdhw7qkka3d31si7k-bash-5.1-p8/bin/bash
make  all-recursive
make[1]: Entering directory '/build/xf86-video-vmware-13.3.0'
Making all in man
make[2]: Entering directory '/build/xf86-video-vmware-13.3.0/man'
  GEN      vmware.4
make[2]: Leaving directory '/build/xf86-video-vmware-13.3.0/man'
Making all in saa
make[2]: Entering directory '/build/xf86-video-vmware-13.3.0/saa'
  CC       libsaa_la-saa.lo
  CC       libsaa_la-saa_pixmap.lo
  CC       libsaa_la-saa_unaccel.lo
  CC       libsaa_la-saa_render.lo
  CC       libsaa_la-saa_accel.lo
  CCLD     libsaa.la
ar: `u' modifier ignored since `D' is the default (see `U')
make[2]: Leaving directory '/build/xf86-video-vmware-13.3.0/saa'
Making all in vmwgfx
make[2]: Entering directory '/build/xf86-video-vmware-13.3.0/vmwgfx'
  CC       libvmwgfx_la-vmwgfx_driver.lo
  CC       libvmwgfx_la-vmwgfx_crtc.lo
  CC       libvmwgfx_la-vmwgfx_output.lo
  CC       libvmwgfx_la-vmwgfx_dri2.lo
  CC       libvmwgfx_la-vmwgfx_tex_video.lo
  CC       libvmwgfx_la-vmwgfx_saa.lo
  CC       libvmwgfx_la-vmwgfx_drmi.lo
In file included from /nix/store/g405x9n50fff4aks50am85g39n6w82k3-libdrm-2.4.108-dev/include/xf86drmMode.h:45,
                 from vmwgfx_saa.c:32:
xorg-server-1.20.13/include/xorg/xf86Opt.h:44:10: error: two or more data types in declaration specifiers
In file included from xorg-server-1.20.13/include/xorg/xf86str.h:43,
                 from xorg-server-1.20.13/include/xorg/xf86.h:44,
                 from ../saa/saa.h:43,
                 from vmwgfx_saa.h:31,
                 from vmwgfx_saa.c:34:
xorg-server-1.20.13/include/xorg/xf86Opt.h:44:14: warning: declaration does not declare anything
  CC       libvmwgfx_la-vmwgfx_overlay.lo
In file included from /nix/store/g405x9n50fff4aks50am85g39n6w82k3-libdrm-2.4.108-dev/include/xf86drmMode.h:45,
                 from vmwgfx_driver.h:41,
                 from vmwgfx_tex_video.c:32:
xorg-server-1.20.13/include/xorg/xf86Opt.h:44:10: error: two or more data types in declaration specifiers
In file included from xorg-server-1.20.13/include/xorg/xf86str.h:43,
                 from xorg-server-1.20.13/include/xorg/xf86.h:44,
                 from vmwgfx_driver.h:43,
                 from vmwgfx_tex_video.c:32:
xorg-server-1.20.13/include/xorg/xf86Opt.h:44:14: warning: declaration does not declare anything
make[2]: *** [Makefile:508: libvmwgfx_la-vmwgfx_saa.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [Makefile:501: libvmwgfx_la-vmwgfx_tex_video.lo] Error 1
In file included from /nix/store/g405x9n50fff4aks50am85g39n6w82k3-libdrm-2.4.108-dev/include/xf86drmMode.h:45,
                 from vmwgfx_driver.h:41,
                 from vmwgfx_drmi.c:46:
xorg-server-1.20.13/include/xorg/xf86Opt.h:44:10: error: two or more data types in declaration specifiers
In file included from xorg-server-1.20.13/include/xorg/xf86str.h:43,
                 from xorg-server-1.20.13/include/xorg/xf86.h:44,
                 from vmwgfx_driver.h:43,
                 from vmwgfx_drmi.c:46:
xorg-server-1.20.13/include/xorg/xf86Opt.h:44:14: warning: declaration does not declare anything
make[2]: *** [Makefile:515: libvmwgfx_la-vmwgfx_drmi.lo] Error 1
make[2]: Leaving directory '/build/xf86-video-vmware-13.3.0/vmwgfx'
make[1]: *** [Makefile:449: all-recursive] Error 1
make[1]: Leaving directory '/build/xf86-video-vmware-13.3.0'
make: *** [Makefile:381: all] Error 2
builder for '/nix/store/vpa89ysd4r467s0ry4rq2a1kg9qx17b0-xf86-video-vmware-13.3.0.drv' failed with exit code 2
error: build of '/nix/store/vpa89ysd4r467s0ry4rq2a1kg9qx17b0-xf86-video-vmware-13.3.0.drv' failed
```

This breaks a NixOS test which is blocking `nixos-unstable`'s advance.

See also [hydra build](https://hydra.nixos.org/build/158733535) and [failing test build](https://hydra.nixos.org/build/159581161).

### Steps To Reproduce

`nix-build -A xorg.xf86videovmware` on `x86_64-linux`.

--- a/pkgs/servers/x11/xorg/overrides.nix
+++ b/pkgs/servers/x11/xorg/overrides.nix
@@ -648,6 +648,14 @@ self: super:
           #
           # We set it to /var/log which can't be touched from inside the sandbox causing the build to hard-fail
           ./dont-create-logdir-during-build.patch
+
+          # Fix e.g. xorg.xf86videovmware with libdrm 2.4.108
+          # TODO: remove with xorgserver >= 1.21
+          (fetchpatch {
+            name = "stdbool.patch";
+            url = "https://gitlab.freedesktop.org/xorg/xserver/-/commit/454b3a826edb5fc6d0fea3a9cfd1a5e8fc568747.diff";
+            sha256 = "1l9qg905jvlw3r0kx4xfw5m12pbs0782v2g3267d1m6q4m6fj1zy";
+          })
         ];
         buildInputs = commonBuildInputs ++ [ libdrm mesa ];
         propagatedBuildInputs = attrs.propagatedBuildInputs or [] ++ [ libpciaccess libepoxy ] ++ commonPropagatedBuildInputs ++ lib.optionals stdenv.isLinux [

--- a/pkgs/servers/x11/xorg/overrides.nix
+++ b/pkgs/servers/x11/xorg/overrides.nix
@@ -452,6 +452,16 @@ self: super:
   });
 
   xf86videoqxl = super.xf86videoqxl.overrideAttrs (attrs: {
+    # https://gitlab.freedesktop.org/xorg/driver/xf86-video-qxl/-/issues/12
+    postPatch = ''
+      patch -p1 <<EOF
+      --- a/src/qxl_option_helpers.c
+      +++ b/src/qxl_option_helpers.c
+      @@ -37 +37 @@
+      -        return options[option_index].value.bool;
+      +        return options[option_index].value.boolean;
+      EOF
+    '';
     buildInputs =  attrs.buildInputs ++ [ spice-protocol ];
   });
 

